tools/ansible/tasks/wsu/test/e2e/wsu_test.go

```diff
@@ -0,0 +1,122 @@
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir            = os.Getenv("ARTIFACT_DIR")
+	privateKeyPath = os.Getenv("KUBE_SSH_KEY_PATH")
+
+	// Path of the WSU playbook
+	playbookPath = os.Getenv("WSU_PATH")
+	// clusterAddress is the address of the OpenShift cluster e.g. "foo.fah.com".
+	// This should not include "https://api-" or a port
+	clusterAddress = os.Getenv("CLUSTER_ADDR")
+
+	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
+	// Microsoft Windows Server 2019 Base with Containers.
+	imageID      = "ami-0b8d82dea356226d3"
+	instanceType = "m4.large"
+	sshKey       = "openshift-dev"
+
+	// Cloud provider factory that we will use in these tests
+	cloud cloudprovider.Cloud
+	// Credentials for a spun up instance
+	createdInstanceCreds *types.Credentials
+	// Temp directory ansible created on the windows host
+	ansibleTmpDir = ""
+)
+
+// createAWSWindowsInstance creates a windows instance and populates the "cloud" and "createdInstanceCreds" global
+// variables
+func createAWSWindowsInstance(t *testing.T) {
+	var err error
+	cloud, err = cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
+		imageID, instanceType, sshKey, privateKeyPath)
+	require.NoErrorf(t, err, "Error setting up cloud provider: %s", err)
+	createdInstanceCreds, err = cloud.CreateWindowsVM()
+	require.NoErrorf(t, err, "Error spinning up Windows VM: %v", err)
+	require.NotNil(t, createdInstanceCreds, "Returned instance credentials are empty")
+}
+
+// createhostFile creates an ansible host file and returns the path of it
+func createHostFile(t *testing.T, ip, password string) string {
+	hostFile, err := ioutil.TempFile("", "testWSU")
+	defer hostFile.Close()
+	require.NoErrorf(t, err, "Could not make temporary file: %s", err)
+
+	_, err = hostFile.WriteString(fmt.Sprintf(`[win]
+%s ansible_password='%s'
+
+[win:vars]
+ansible_user=Administrator
+cluster_address=%s
+ansible_port=5986
+ansible_connection=winrm
+ansible_winrm_server_cert_validation=ignore`, ip, password, clusterAddress))
+	require.NoError(t, err, "Could not write to host file: %s", err)
+	return hostFile.Name()
+}
+
+// TestWSU creates a Windows instance, runs the WSU, and then runs a series of tests to ensure all expected
+// behavior was achieved
+func TestWSU(t *testing.T) {
+	// TODO: Check if other cloud provider credentials are available
+	if awscredentials != "" {
+		createAWSWindowsInstance(t)
+	} else {
+		t.Fatal("No cloud provider credentials available")
+	}
+	defer cloud.DestroyWindowsVMs()
+	// In order to run the ansible playbook we create an inventory file:
+	// https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+	hostFilePath := createHostFile(t, createdInstanceCreds.GetIPAddress(), createdInstanceCreds.GetPassword())
```

Comment:
Can't we provide the info as command line inputs instead of the host file?

Reply:
I thought it would be handy to do it via a hostfile, as it makes running the tests on mutiple hosts simple, if we wanted to do that in the future

Reply:
And we can't do that with command line args? Anyhow I won't insist on this change.